gcc/rust/checks/errors/borrowck/polonius/rust-polonius-ffi.h

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2020-2023 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef RUST_POLONIUS_FACTS_FFI_H
+#define RUST_POLONIUS_FACTS_FFI_H
+
+// This file defines the C++ side of the FFI interface to Polonius.
+// The corresponding Rust side is in `gccrs-ffi.rs`.
+
+// IMPORTANT:
+// This file intentionally does not include any C++ headers
+// to allow seamless binding generation on the Rust side.
+
+#include <cstdint>
+
+namespace Rust {
+namespace Polonius {
+
+using Origin = size_t;
+using Loan = size_t;
+using Point = size_t;
+using Variable = size_t;
+using Path = size_t;
+
+namespace FFI {
+
+// NOTE: std::pair and std::tuple are complicating the bindings' generation.
```

Comment:
yes, this is something @P-E-P and I have also run into when making FFI types/interfaces with Rust bits. we should probably start keeping a library of C++ FFI types for Rust in `gcc/rust/util` - things like `Option`, `Pair`, `Tuple`, `String`, `Str` etc

Reply:
Couldn't we make a new library in libgrust instead ? I'm sure those FFI types could benefits the proc macros.

Reply:
> Couldn't we make a new library in libgrust instead ? I'm sure those FFI types could benefits the proc macros.

Sounds good to me.

Reply:
Moved to #2923